### PR TITLE
add the test plan to our docs

### PR DESCRIPTION
**Pull request by m8ttyB**
Initial push of the test plan docs. Note, this was heavily reviewed and iterated on in [Google Docs](https://docs.google.com/a/mozilla.com/document/d/1-fNb78mMcUIs9UMVs0qnAa5lAK_47kOGPKz4ruzDJwU/edit?usp=sharing).

Completes one of the requirements for issue #218

### CODE REVIEW

File: docs/test-plan.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+* All test cases have been executed
+
+* A sufficient level of automated test coverage has been achieved (measured by code coverage tools). Test coverage will also be measured against reduction of risk in the areas designated by the risk assessment plan.
+
+* All blockers, criticals must be fixed and verified or have an agreed-upon timeline for being fixed (as determined by engineering/QA)
+
+### Checklist
+
+<table>
```
Comment (linuxwolf) [inline docs/test-plan.md]: This works, but adding awareness that (Github flavored Markdown and) MkDocs support a more concise table definition:

http://www.mkdocs.org/user-guide/writing-your-docs/#tables

File: docs/test-plan.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+* All telemetry requirements are in place
+* All the necessary PR and blog posts have been sent out 
+
+## Overall Status
+
+Alpha - estimated Q3 timeline
+
+* July: lay the foundation for system add-on with embedded WebExtension development
+* August: begin secure storage and front-end editor design and implementation
+* September: finish editor and storage, implement lock/unlock, package for release
```
Comment (linuxwolf) [inline docs/test-plan.md]: Two of these were implemented in a way users would experience them.  I would say that lock/unlock is partially complete from a user's perception.

File: docs/test-plan.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+Beta - estimated Q4 timeline
+
+* September: finish Firefox extension storage, front-end editor, and messaging
+* October: integrate/fork Firefox password save/fill mechanics
+* November: implement sync, mobile apps, and "first run" experiences
```
Comment (linuxwolf) [inline docs/test-plan.md]: Need to double-check with PM, but I think mobile apps is now pushed off until 2018Q1

File: docs/test-plan.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+Alpha - estimated Q3 timeline
+
+* July: lay the foundation for system add-on with embedded WebExtension development
+* August: begin secure storage and front-end editor design and implementation
+* September: finish editor and storage, implement lock/unlock, package for release
+
+Beta - estimated Q4 timeline
+
+* September: finish Firefox extension storage, front-end editor, and messaging
+* October: integrate/fork Firefox password save/fill mechanics
```
Comment (linuxwolf) [inline docs/test-plan.md]: This is not accurate -- fill will not be touched in October.  I think when this happens is still very fluid.

File: docs/test-plan.md (hunk 3 above)
Comment (linuxwolf) [reply docs/test-plan.md]: Further, sync and a fully realized first-time experience is still being fleshed out.  I think the "final" first-time experience will happen in November, but I'm doubtful about synchronization given some of the idP hurdles we still need to overcome.

File: docs/test-plan.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+* QA has access to all PRDs, mocks, and related documents.
+* A run script that installs the WebExtension for Firefox Nightly
+* A compiled and signed extension for Firefox Beta and Release
+* Staging and release channels
+
+### Exit Criteria
+
+* All bugs against the features have been triaged
+* All P1 and P2 bugs have been resolved fixed and verified by Product Integrity
```
Comment (pdehaan) [inline docs/test-plan.md]: Are we tagging issues w/ priorities? Or should this just say all [**TxP**](https://github.com/mozilla-lockbox/lockbox-extension/issues?q=is%3Aopen+is%3Aissue+label%3ATxP) tagged issues?

File: docs/test-plan.md (hunk 2 above)
Comment (m8ttyB) [reply docs/test-plan.md]: r+ Changed section header to `Timelines and Milestones`. Also removed lock/unlock.

File: docs/test-plan.md (hunk 4 above)
Comment (m8ttyB) [reply docs/test-plan.md]: Updated test plan to reflect that milestones and tracking of user stories is taking place in Github

File: docs/test-plan.md (hunk 5 above)
Comment (m8ttyB) [reply docs/test-plan.md]: Good catch, when this doc was authored, TxP did not exist.

File: docs/test-plan.md (hunk 1 above)
Comment (m8ttyB) [reply docs/test-plan.md]: Nice, I did not know of this thing. Done!

**Pull request merged**